solid/solid-file-client/solid.js

```diff
@@ -226,31 +210,31 @@ function getSolidDocumentUrl(resourceUrl) {
 }
 
 async function fetchUserProfile(webId) {
-    const profileQuads = await readSolidDocument(webId);
-    const nameQuad = profileQuads.find(quad => quad.predicate.value === 'http://xmlns.com/foaf/0.1/name');
-    const storageQuad = profileQuads.find(quad => quad.predicate.value === 'http://www.w3.org/ns/pim/space#storage');
-
+    const [nameQuad] = await readSolidDocument(webId, null, { foaf: 'name' })
+    await findPodStorage(webId)
     return {
         url: webId,
-        name: nameQuad?.object.value || 'Anonymous',
-        storageUrl: storageQuad?.object.value || await findUserStorage(webId),
+        name: nameQuad?.object.value || webId.split('//')[1].split('/profile')[0], // replace 'Anonymous',
+        storageUrl: await findPodStorage(webId),
     };
 }
 
-async function findUserStorage(url) {
+async function findPodStorage(url) {
     url = url.replace(/#.*$/, '');
     url = url.endsWith('/') ? url + '../' : url + '/../';
     url = new URL(url);
 
-    const response = await solidFileClient.fetch(url.href);
+    // following solid/protocol used by NSS and CSS
+    const [StorageQuad] = await readSolidDocument(url.href, null, null, { space: 'Storage' })
 
-    if (response.headers.get('Link')?.includes('<http://www.w3.org/ns/pim/space#Storage>; rel="type"'))
-        return url.href;
+    if (StorageQuad) {
+        return url.href
+    }
 
-    if (url.pathname === '/')
-        return url.href;
+    // for old NSS not following storage discovery protocol
```

Comment:
I don't think it's necessary to mention NSS in particular here, if you want to document this we can just say "for providers that don't advertise storage properly" or something like that. There may be servers other than old NSS doing this.

Reply:
good